### PR TITLE
Adopt Okio's FileSystem in Wire

### PR DESCRIPTION
**Pull request by swankjesse**
The biggest potential upside here is being able to perform file
loading in multiplatform code.

### CODE REVIEW

File: wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt [hunk 1]
```diff
@@ -240,11 +244,21 @@ class WireCompiler internal constructor(
       }
     }
 
+    @Throws(WireException::class)
+    @JvmStatic
+    fun forArgs(
```
Comment (oldergod) [inline wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:249]: 👍

File: wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/fileSystems.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/*
```
Comment (oldergod) [inline wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/fileSystems.kt:1]: Just confirming you want the file name to start with a lower case

**Review by oldergod — APPROVED** [review]

File: wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/fileSystems.kt (hunk 2 above)
Comment (swankjesse) [reply wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/fileSystems.kt:1]: Yeah, I think the scheme I observed elsewhere is lowerCamel – no primary top-level class, UpperCamel – primary top-level class

**Review by oldergod — APPROVED** [review]

**Comment by blacktooth** [comment]
Looks like this change prevents use of Jimfs for in-memory schema loading use-cases. Is it intentional? 

Can [FakeFileSystem](https://github.com/square/okio/blob/master/okio-fakefilesystem/src/commonMain/kotlin/okio/fakefilesystem/FakeFileSystem.kt) be reliably used for production use-cases where schemas will be loaded in-memory?

**Pull request merged**